POD breakage in warnings.pm

openstrike opened the pull request:
An unescaped right angle bracket was resulting in a malformed code snippet in warnings.pm

This tiny PR escapes that character and thus fixes the rendering.

openstrike committed:
POD breakage in warnings.pm

An unescaped right angle bracket was resulting in a malformed code snippet in warnings.pm

This tiny PR escapes that character and thus fixes the rendering.

xenu commented:
Well spotted! However, the comments at the top of `warnings.pm` say:

```
# !!!!!!!   DO NOT EDIT THIS FILE   !!!!!!!
# This file is built by regen/warnings.pl.
# Any changes made here will be lost!
```

You should make that change in `regen/warnings.pl` instead.

ilmari commented:
As noted at the top of `lib/warnings.pm`:
```
# -*- buffer-read-only: t -*-
# !!!!!!!   DO NOT EDIT THIS FILE   !!!!!!!
# This file is built by regen/warnings.pl.
# Any changes made here will be lost!
```
Please make the change in `regen/warnings.pl` instead, and run that to update `lib/warnings.pm`.
Also, it's more readable to use the multi-bracket syntax  `C<< … >>` instead of escaping `<` and `>`, as seen in the next paragraph.

openstrike committed:
Apply fix to warnings.pl instead

openstrike commented:
Thanks for letting me know. I had not considered that an auto-generated file would be in the repo like that (and obviously didn't read the first few lines). I have added the change to warnings.pl now (and switched to the double-bracket form there).

ilmari commented:
Thanks for the updated patch
However, there are a few issues with it still:
1. The `C<< … >>` syntax requires space inside the brackets
2. Please run `perl regen/warnings.pl` to update `lib/warnings.pm`
3. Please verify that the regenerated POD renders correctly using `perldoc lib/warnings.pm`

openstrike committed:
Further amendment of fix to warnings.pl

openstrike commented:
Gah. That should show that I never use the `C<< … >>` form.

Hopefully f03f693 is it fully fixed now.

ilmari commented:
@openstrike A few of the porting tests failed:

```
Pete Houston  <githubdevteam@openstrike.co.uk> not found in AUTHORS
porting/authors.t .................................................... 
Failed 1/2 subtests 
# not ok 9 - lib/warnings.pm version 1.55
porting/cmp_version.t ................................................ 
Failed 1/9 subtests
```
Please add yourself to the `AUTHORS` file and bump the `$VERSION` in `regen/warnings.pl` and rerun it to update `lib/warnings.pm`.

openstrike committed:
Housekeeping (version bump, AUTHORS entry)

openstrike commented:
Those amendments are now in place. There are now 4 individual tests failing for cygwin only - I am struggling to see how these might be related to this PR, to be honest. If it needs something further from me, please say.

jkeenan commented:
Manually rebased; commits squashed; merged to blead in a9d7a07c2e.

Closing p.r., thanks for your contribution.
